Addon: add headlamp (#7)
kinvolk/minikube: deploy/addons/headlamp/README.md

> ## Headlamp Addon

Reviewer: Add a newline under this.

> [Headlamp](https://github.com/kinvolk/headlamp) - Headlamp is an easy-to-use and extensible Kubernetes web UI.

Author: Have added documentation for the addon.

Reviewer: BTW, why not just add the link + the description: [Headlamp](...) is an easy-to-use ....

Reviewer: I have also noticed that apparently other addons don't have a README.md file, so not sure if they will accept it for us.

> [Headlamp](https://github.com/kinvolk/headlamp) - Headlamp is an easy-to-use and extensible Kubernetes web UI.

Reviewer: ```suggestion
[Headlamp](https://kinvolk.github.io/headlamp) - Headlamp is an easy-to-use and extensible Kubernetes web UI.
```

### Enable Headlamp on minikube

To enable this addon, simply run:
```shell script
minikube addons enable headlamp
```

> Once the addon is enable you can access the Headlamp web UI use the following command.

Reviewer: ```suggestion
Once the addon is enabled, you can access Headlamp's web UI using the following command:
```

```shell script
minikube service headlamp -n headlamp
```

> To authenticate in Headlamp, fetch the Authentication Token use the following command:

Reviewer: ```suggestion
To authenticate in Headlamp, fetch the Authentication Token using the following command:
```

```shell script
export SECRET=$(kubectl get secrets --namespace headlamp -o custom-columns=":metadata.name" | grep "headlamp-token")
kubectl get secret $SECRET --namespace headlamp --template=\{\{.data.token\}\} | base64 --decode
``` 

Headlamp can display more detailed information when metrics-server is installed. To install it, run:

```shell script
minikube addons enable metrics-server	
```		

### Testing installation

```shell script

> kubectl get po -n headlamp

Reviewer: Should it be `get pods` (to be a bit more formal).

```

> If everything went well you shouldn't get any errors about headlamp being installed in your cluster.

Reviewer: ```suggestion
If everything went well, there should be no errors about Headlamp's installation in your Minikube cluster.
```

### Disable headlamp

> To disable this addon, simply run:

Reviewer: Add a newline under each of these lines.
